lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -125,8 +125,12 @@ public Channel get(String channelName) {
 		public Channel get(String channelName, ChannelOptions channelOptions) throws AblyException {
 			Channel channel = super.get(channelName);
 			if (channel != null) {
-				if (channelOptions != null)
+				if (channelOptions != null) {
+					if (channel.shouldReattachToSetOptions(channelOptions)) {
+						throw AblyException.fromErrorInfo(new ErrorInfo("Channels.get() cannot be used to set channel options that would cause the channel to reattach. Please, use Channel.setOptions() instead.", 40000, 400));
```

Comment:
Did we agree that this would have a unique error code so that we can have a help article for this?